pinot-controller/src/main/java/org/apache/pinot/controller/util/TableMetadataReader.java

```diff
@@ -60,21 +60,28 @@ public TableMetadataReader(Executor executor, HttpClientConnectionManager connec
 
   public Map<String, JsonNode> getServerCheckSegmentsReloadMetadata(String tableNameWithType, int timeoutMs)
       throws InvalidConfigException, IOException {
+    List<String> segmentsMetadata = getIsReloadNeeded(tableNameWithType, timeoutMs);
+    Map<String, JsonNode> response = new HashMap<>();
+    for (String segmentMetadata : segmentsMetadata) {
+      JsonNode responseJson = JsonUtils.stringToJsonNode(segmentMetadata);
+      response.put(responseJson.get("instanceId").asText(), responseJson);
+    }
+    return response;
+  }
+
+  private List<String> getIsReloadNeeded(String tableNameWithType, int timeoutMs)
```

Comment:
```suggestion
  private List<String> getReloadCheckResponses(String tableNameWithType, int timeoutMs)
```

Reply:
Do you want to add a new method boolean isReloadNeeded()?
Re: I am thinking to extract boolean in the integration with the server responses.